src/SFA.DAS.EAS.Web/Orchestrators/EmployerAgreementOrchestrator.cs

```diff
@@ -171,17 +171,12 @@ public EmployerAgreementOrchestrator(IMediator mediator, ILog logger, EmployerAp
         }
 
         public async Task<OrchestratorResponse> SignAgreement(string agreementid, string hashedId, string externalUserId,
-            DateTime signedDate)
+            DateTime signedDate, string companyName)
```

Comment:
might not be a company, and might be better to add the legal entity id so that if we have a charity and company with the same name we know which one it is.

Reply:
We need to be specific. I am using it to construct a sentence. It could be argued that messages need to pass every actor as separate properties in case some processor needs it but I'm being Agile here.. just doing what I need to and being specific

Reply:
I'm confused as you name it as ProviderName in the message. So it can't be specific to a company if its called provider in the message. Are you using this as a way to persist the data so that if the legal entity changes we still have the historic name? Are these details not something we can attached to the agreement model itself if it needs to be persisted?
Otherwise we aren't we just sending all the data that everyone needs into the message? This would just bloat the message out.

Reply:
The reply above related to the message comments below about the persist nature of messages. This is why I'm raising the concern mainly as we don't use messages at present to persist data states. Not to my knowledge at least. which is why Ids are used to allow listeners of messages to collect all the data they need that relates to the message and then they can handle the persistence.

Reply:
I've replaced companyName with organisationName in these messages. It is the case though that we need strings; names etc not ids. Michael confirms.